config/sidekiq.yml

```diff
@@ -1,6 +1,8 @@
-# Sample configuration file for Sidekiq.
+# Configuration file for Sidekiq (but not when running in Kubernetes).
+#
+# This file is NOT used in EKS/Replatforming. See sidekiq_k8s.yml.
```

Comment:
I think we should try avoid having a separate file for this for simplicity.

We can actually use embedded Ruby in this file to reference environment variables.

What I'd actually suggest for this is that we set up an environment variable in puppet to specify the logfile and then have everywhere else log to STDOUT. This is because it's mostly annoying having the logfile in dev (where we've fought it in the past: https://github.com/alphagov/content-publisher/blob/3e2ad9d23852bbc691d52922283a33d2dbc9acb9/config/sidekiq.yml#L5-L7)

So I'd suggest having the file set-up like:

```
:logfile: <%= ENV.fetch("SIDEKIQ_LOGFILE", "~") %> # default to stdout, unless logfile specified
```

and then adding that env var to: https://github.com/alphagov/govuk-puppet/blob/5e3be50fee026d9d33d9c7f68c2251262573b48c/modules/govuk/manifests/deploy/config.pp#L121-L134 so all apps can pick it up (note: this app seems to be a little unconventional by not having ./log/sidekiq.json.log as a name which is the convention: https://github.com/alphagov/govuk-puppet/blob/5b5e249600eeddbd78e5292bdf4a8999c8b35a26/modules/govuk/manifests/app.pp#L442-L447)

Reply:
Ooh ok, hadn't spotted that it puts the file through ERB :D Sounds great - let's go with that.

Reply:
RFAL. I wrapped it in a conditional, for forward-compatibility with Sidekiq 6. Puppet PR on the way.

Reply:
(I considered renaming the file to make it play nice with editor syntax plugins, but thought it safer not to for now since the filename is part of the command in the Procfile etc.)

Reply:
Puppet PR: https://github.com/alphagov/govuk-puppet/pull/11673